src/group_actions/group_action_interface.jl

```diff
@@ -0,0 +1,213 @@
+with the following properties
+
+1. ``σ(p, $(_math(:e))) = p`` holds for all ``p ∈ $(_math(:M))``
+2. ``σ(σ(p, g), h) = σ(p, g$(_math(:∘))h)`` holds for all ``g,h ∈ $(_math(:G))``, ``p ∈ $(_math(:M))``
+
+where ``$(_math(:∘))`` denotes the group operation of the [`LieGroup`](@ref) ``$(_math(:G))``.
+See also [HilgertNeeb:2012; Remark 9.1.12](@cite).
+
+The type of action can be seen a bit better when writing the action as a family ``σ_g(p)``:
+we obtain from the second property as
+
+```math
+  σ_g(σ_h(p)) = σ_{hg}(p)
+```
+
+and see that ``g`` appears on the right.
+
+To emphasize the side the group operation is acting from, we sometimes write ``σ^{$(_tex(:rm,"R"))}``.
+If the action is clear from context we write ``σ(p, g) = p $(_math(:act)) g``.
+
+One notable example of a right action is the inverse of an action of  [`AbstractLeftGroupActionType`](@ref) ``σ^{$(_tex(:rm,"L"))}``,
+which is given by ``τ_g = (σ^{$(_tex(:rm,"L"))}_g)^{-1} = σ^{$(_tex(:rm,"L"))}_{g^{-1}}``.
+We obtain
+
+$(_note(:LeftInverseActionIsRight))
+
+$(_note_action_argument_order)
+"""
+abstract type AbstractRightGroupActionType <: AbstractGroupActionType end
+
+@doc """
+    AbstractGroupAction{T<:AbstractGroupActionType, L<:LieGroup, M<:AbstractManifold}
+
+An abstract group action of [`AbstractGroupActionType`](@ref) `T` of a [`LieGroup`](@ref) of type `L`
+acting on an $(_link(:AbstractManifold)) of type `M`.
+
+See [HilgertNeeb:2012; Section 9.1.3](@cite) for more details.
+"""
+abstract type AbstractGroupAction{
+    T<:AbstractGroupActionType,L<:LieGroup,M<:AbstractManifold
+} end
+
+function base_lie_group end
+@doc """
+    base_Lie_group(A::AbstractGroupAction)
+
+Return the [`LieGroup`](@ref) of the [`AbstractGroupAction`](@ref).
+"""
+base_lie_group(::AbstractGroupAction)
+
+@doc """
+    base_manifold(A::AbstractGroupAction)
+
+Return the $(_link(:AbstractManifold)) of the [`AbstractGroupAction`](@ref).
+"""
+ManifoldsBase.base_manifold(::AbstractGroupAction)
+
+#
+#
+# Functions
+
+_doc_apply = """
+    apply(A::AbstractGroupAction{T, L, M}, g, p)
+    apply!(A::AbstractGroupAction{T, L, M}, q, g, p)
+
+Apply the group action induced by ``g ∈ $(_math(:M))`` to ``p ∈ $(_math(:M))``,
+where the kind of group action is indicated by the [`AbstractGroupActionType`](@ref) `T`.
+This can be perfomed in-place of `q`.
+"""
+
+# function apply end # uncomment (remove this comment) when removing this function from Manifolds.jl
+@doc "$(_doc_apply)"
+function apply(A::AbstractGroupAction, g, p)
+    q = allocate_result(A, apply, g, p)
+    apply!(A, q, g, p)
+    return q
+end
+
+# function apply! end # uncomment (remove this comment) when removing this function from Manifolds.jl
+@doc "$(_doc_apply)"
+apply!(A::AbstractGroupAction, q, g, p)
+
+_doc_diff_apply = """
+    diff_apply(A::AbstractGroupAction{T, L, M}, g, p, X)
+    diff_apply!(A::AbstractGroupAction{T, L, M}, Y, g, p, X)
+
+Compute the differential ``Dσ_g(p): T_p$(_math(:M)) → T_{σ_g(p)}$(_math(:M))``,
```

Comment:
```suggestion
Compute the differential ``D_{_math(:M)} σ_g(p): T_p$(_math(:M)) → T_{σ_g(p)}$(_math(:M))``,
```
I'd suggest using different symbols for differential wrt. manifold and group.

Reply:
Is this the best notation? You would use $D_{\mathcal G}$ for the group one?

Reply:
One could also maybe use $D_g$ and $D_p$ indicating the variable we would take the diff of?

Reply:
Sure, $D_g$ and $D_p$ are also fine.

Reply:
I prefer that one than but will also introduce a symbol for that then.